service/src/test/java/greencity/filters/HabitFactSpecificationTest.java

```diff
@@ -0,0 +1,236 @@
+package greencity.filters;
+
+import greencity.entity.*;
+import jakarta.persistence.criteria.*;
+import jakarta.persistence.metamodel.SingularAttribute;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+
+@ExtendWith(MockitoExtension.class)
+class HabitFactSpecificationTest {
+
+    @Mock
+    private Root<HabitFact> HabitFactRoot;
+
+    @Mock
+    private CriteriaBuilder builder;
+
+    @Mock
+    private CriteriaQuery MockCriteriaQuery;
+
+    @Mock
+    private Path<Object> id;
+
+    @Mock
+    private Path<Long> habitIdPath;
+
+    @Mock
+    private Path<Long> habitFactRootIdPath;
+
+    @Mock
+    private Join<HabitFact,Habit> habitJoin;
+
+    @Mock
+    private Path<String> contentPath;
+
+    @Mock
+    private Predicate likePredicate;
+
+    @Mock
+    private Expression<String> as;
+    HabitFactSpecification specification;
+
+    @Test
+    void toPredicate_Id(){
+        List<SearchCriteria> searchCriteriaList = new ArrayList<>();
+        searchCriteriaList.add(SearchCriteria.builder()
+                .key("id")
+                .type("id")
+                .value(2L)
+                .build());
```

Comment:
minor nitpick: use `List.of`
```suggestion
        List<SearchCriteria> searchCriteriaList = List.of(
                SearchCriteria.builder()
                        .key("id")
                        .type("id")
                        .value(2L)
                        .build());
```